composer/trainer/trainer.py

```diff
@@ -1251,6 +1335,7 @@ def __init__(
         if parallelism_config is not None:
             # Patch PyTorch to fix distributed bugs
             patch_pytorch()
+            patch_unshard_for_automicrobatching(False)
```

Comment:
why does this patch need to be separate? can we not just include in pytorch patch? concerned this adds more maintenance overhead.

Reply:
this patch is dynamic, since it flips between the original FSDP implementation of unshard and our patched unshard with the sync live during the training loop - it wouldn't make sense to call patch_pytorch() and not change 90% of the methods. in terms of maintenance overhead, regardless of whether it's in `patch_pytorch` or `patch_unshard_for_automicrobatching`, theres still going to be exactly one method that needs to be maintained (`unshard_with_sync`)